### PR TITLE
Handle and log rejection of socket close

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

ADO item: https://dev.azure.com/fluidframework/internal/_workitems/edit/3648
Handle and log rejection of socket close. This along with other Prs:
Pr1: https://github.com/microsoft/FluidFramework/pull/14021
Pr2: https://github.com/microsoft/FluidFramework/pull/14350
should solve the bug and if not then give us more insights into what is happening.

### CODE REVIEW

**Review by NicholasCouri — APPROVED** [review]

File: packages/drivers/driver-base/src/documentDeltaConnection.ts [hunk 1]
```diff
@@ -472,14 +454,34 @@ export class DocumentDeltaConnection
...
+					const normalizedError = this.addPropsToError(failError);
+					this.logger.sendErrorEvent({ eventName: "CloseSocketError" }, normalizedError);
+				}
 				reject(err);
 			};
 
 			const failConnection = (err: IAnyDriverError) => {
-				this.disconnect(err);
+				try {
+					this.disconnect(err);
```
Comment (vladsud) [inline packages/drivers/driver-base/src/documentDeltaConnection.ts:468]: I'd hope that this is not required, as Container (and probably DM) will use EventEmitter that captures exceptions in handlers. Or do we think that there is an excetpion in our code processing disconnect?

File: packages/drivers/driver-base/src/documentDeltaConnection.ts [hunk 2]
```diff
@@ -629,16 +631,25 @@ export class DocumentDeltaConnection
 			});
 
 			this.socket.emit("connect_document", connectMessage);
-
```
[the remark this answers is not in the document]
Comment (jatgarg) [reply packages/drivers/driver-base/src/documentDeltaConnection.ts:632]: Yes

File: packages/drivers/driver-base/src/documentDeltaConnection.ts (hunk 1 above)
Comment (jatgarg) [reply packages/drivers/driver-base/src/documentDeltaConnection.ts:468]: assert 0x416 hits during closeSocket. So, I want to log that and other state when that happens. Other PR: https://github.com/microsoft/FluidFramework/pull/14021 should prevent 0x416 from happening but I want to add more telemetry as it is affecting a good number of loop users.

File: packages/drivers/driver-base/src/documentDeltaConnection.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jatgarg) [reply packages/drivers/driver-base/src/documentDeltaConnection.ts:632]: I just moved it above before we set the listeners.

**Review by vladsud — APPROVED** [review]

**Pull request merged**